### ISSUE TITLE
FR: implement rand() for tuple types.

### ISSUE THREAD

**Issue by gafter** [JuliaLang/julia#50236]
It would be nice to be able to generate, for example, a random value of type `Tuple{Int, Bool}` without having to implement it myself.  Something like the following:

```julia
function Random.rand(rng::AbstractRNG, ::Type{A}) where {A <: Tuple}
    return ((Random.rand(rng, a) for a in A.parameters)...,)
end
```

**Comment by adienes** [comment]
```
julia> y::Tuple{Int, Bool} = (4, false)
(4, false)

julia> rand.(Ref(rng), typeof.(y))
(-894747688966645945, false)
```

**Comment by giordano** [comment]
Yeah, I'm not sure we really need a new method for this, broadcasting gives you that with very little effort.

**Comment by gafter** [comment]
```
julia> y::Tuple{Tuple{Bool, Bool}, Bool} = ((true, false), true)
((true, false), true)

julia> rand.(Ref(rng), typeof.(y))
ERROR: ArgumentError: Sampler for this object is not defined
Stacktrace:
  [1] Random.Sampler(#unused#::Type{RandomDevice}, sp::Random.SamplerType{Tuple{Bool, Bool}}, #unused#::Val{1})
    @ Random /Applications/Julia-1.8.app/Contents/Resources/julia/share/julia/stdlib/v1.8/Random/src/Random.jl:146
  [2] Random.Sampler(rng::RandomDevice, x::Random.SamplerType{Tuple{Bool, Bool}}, r::Val{1})
    @ Random /Applications/Julia-1.8.app/Contents/Resources/julia/share/julia/stdlib/v1.8/Random/src/Random.jl:140
...
```

**Comment by adienes** [comment]
one place this might be useful is input fuzzing given method signatures


```
for method ∈ methods(foo)
    _sig = method.sig.parameters[2]
    @test foo(rand(_sig)...) == foo_slower_but_correct(rand(_sig)...)
end
```

**Comment by adienes** [comment]
also e.g. sampling points on a hypercube does become a lot easier

```
julia> rand(NTuple{5, Bool}, 10)
10-element Vector{NTuple{5, Bool}}:
 (1, 0, 0, 1, 1)
 (1, 0, 0, 1, 1)
 (1, 0, 0, 0, 1)
 (0, 0, 0, 1, 0)
 (0, 1, 1, 0, 1)
 (0, 0, 1, 0, 0)
 (1, 1, 0, 1, 0)
 (1, 1, 1, 1, 1)
 (1, 1, 0, 0, 0)
 (1, 1, 1, 0, 1)
```

**Comment by stevengj** [comment]
This works:
```jl
julia> rand.((Bool,Bool,Bool))
(true, false, true)

julia> rand.(ntuple(x -> Bool, 10))
(false, true, true, false, true, true, true, false, true, false)
```

**Comment by rfourquet** [comment]
Closed by #50251.
